api/xds_protocol.rst

```diff
@@ -155,45 +231,109 @@ the resources in the :ref:`DiscoveryResponse <envoy_api_msg_DiscoveryResponse>`
+server, which could have a severe performance impact.
 
 Within a stream, new :ref:`DiscoveryRequests <envoy_api_msg_DiscoveryRequest>` supersede any prior
 :ref:`DiscoveryRequests <envoy_api_msg_DiscoveryRequest>` having the same resource type. This means that
 the management server only needs to respond to the latest
 :ref:`DiscoveryRequest <envoy_api_msg_DiscoveryRequest>` on each stream for any given resource type.
 
-Resource hints
-^^^^^^^^^^^^^^
+How the client specifies what resources to return
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+xDS requests allow the client to specify a set of resource names as a hint to the server about which resources the client
+is interested in. In the SotW API variants, this is done via the :ref:`resource_names
+<envoy_api_field_DiscoveryRequest.resource_names>` specified in the :ref:`DiscoveryRequest
+<envoy_api_msg_DiscoveryRequest>`; in the incremental API variants, this is done via the
+:ref:`resource_names_subscribe <envoy_api_field_DeltaDiscoveryRequest.resource_names_subscribe>` and
+:ref:`resource_names_unsubscribe <envoy_api_field_DeltaDiscoveryRequest.resource_names_unsubscribe>` fields in the
+:ref:`DeltaDiscoveryRequest <envoy_api_msg_DeltaDiscoveryRequest>`.
 
-The :ref:`resource_names <envoy_api_field_DiscoveryRequest.resource_names>` specified in the :ref:`DiscoveryRequest
-<envoy_api_msg_DiscoveryRequest>` are a hint about what resources the client is interested in.
+For `Listener` and `Cluster` resource types, Envoy will always set the resource hints to empty, in which case the server
+should use site-specific business logic to determine the full set of resources that the client is interested in, typically
+based on the client's :ref:`node <envoy_api_msg_Core.Node>` identification. However, other xDS clients (such as gRPC) may
+specify explicit LDS/CDS resources as resource hints, for example if they only have a singleton listener and already know
+its name from some out-of-band configuration.
 
-For :ref:`Listener Discovery Service (LDS) <envoy_api_msg_Listener>` and :ref:`Cluster Discovery Service (CDS)
-<envoy_api_msg_Cluster>`, Envoy will always set the resource hints to empty, in which case the server should use
-site-specific business logic to determine the full set of resources that the client is interested in, typically based on
-the client's node identification. However, other xDS clients may specify explicit LDS/CDS resources as resource hints, for
-example if they only have a singleton listener and already know its name from some out-of-band configuration. For EDS/RDS, the
-resource hints are required.
+For other resource types, the resource hints are required. If the field is empty, that means that the client is not
+interested in any resources of the relevant type.
 
 When the resource hints are specified, the management server must supply the requested resources if they exist. The client will
```

Comment:
Do we want to specify what happens when any of the requested resources don't exist? For example:

- management server remembers these resource_names and will send an update when they come online.
- it is an error (but I have not seen examples of returning an error in a DiscoveryResponse)

Reply:
This is specified in the "Knowing When a Requested Resource Does Not Exist" section below.